### ISSUE TITLE
Update libraries and switch to submodules

### ISSUE THREAD

**Issue by adamkewley** [adamkewley/osmv#14]
- Change to using `git submodule`s - it's an extra build step but
  might be able to put it into CMake for less-experienced devs
  
- Update OpenSim to 4.2

  - And re-enable all the currently-disabled stuff (that was
    disabled for 4.1)

- Update GLEW to 2.2 after figuring out what it actually changes

- Update GLM to 0.9.9.8 - seems like a straightforward upgrade

- Update ImGui to 1.82

- Update toml++ to 3.6.1

**Comment by adamkewley** [comment]
Waiting on OpenSim4.2 upgrade, which is proving to be a giant PITA because it has a *much* larger dependency tree than 4.1

**Comment by adamkewley** [comment]
Splitting OpenSim4.2 into a separate issue - it's *so* much different (in terms of the dependencies it pulls in, breaks Xenial builds, etc.) that it's going to be an ongoing upgrade for a while
